packages/client/components/promptResponse/loomExtension.ts

```diff
@@ -0,0 +1,89 @@
+import {Node} from '@tiptap/core'
+import {JSONContent, mergeAttributes} from '@tiptap/react'
+
+const LOOM_REGEX = /^(?:https:\/\/)?(?:www\.)?loom\.com\/share\/[a-zA-Z0-9]*\?.*$/g
+
+export const unfurlLoomLinks = (rawContent: JSONContent): JSONContent => {
+  if (!rawContent.content) {
+    return rawContent
+  }
+
+  const newContent: JSONContent[] = []
+  rawContent.content.forEach((content) => {
+    if (content.type === 'paragraph') {
+      newContent.push(content, ...getLoomNodesForParagraph(content))
+    } else {
+      newContent.push(unfurlLoomLinks(content))
+    }
+  })
+
+  return {
+    type: rawContent.type,
+    content: newContent
+  }
+}
+
+const getLoomNodesForParagraph = (content: JSONContent) => {
+  const distinctLoomLinks: Record<string, string> = {}
```

Comment:
+1 since this is a local var, if i hit enter to create a new paragraph & paste the same link again, it's going to unfurl the video again. is this intended? 

fwiw i saw that slack messages will do a big unfurl for 1 link, 2 small unfurls for 2 links, and they won't unfurl for 3 links. thought that was kinda cool to prevent it from getting too long! (e.g. thumbnail + title for 1, just thumbnail for 2, just links for 3+). i saw loom has the /oembed path for metadata like title/description if you feel like playing with it